### PR TITLE
Use libc_bitflags! for BSD in fcntl.rs

### PR DESCRIPTION
**Pull request by equal-l2**

### CODE REVIEW

**Comment by Susurrus** [comment]
There's a build failure for NetBSD I don't quite understand. Hopefully you can figure it out.

**Comment by equal-l2** [comment]
I have no idea why `libc` is not viable on NetBSD.

**Comment by Susurrus** [comment]
You can easily test this locally by doing `rustup target install x86_64-unknown-netbsd` and then `cargo build --target=x86_64-unknown-netbsd` since it's just a compilation error. Looks like a module import is missing or something on the netbsd target. Let me know if you get stuck!

**Comment by Susurrus** [comment]
@equal-l2 Are you still able to drive this forward? Would love to get this into the next 0.9 release.

File: src/fcntl.rs [hunk 1]
```diff
@@ -404,33 +404,33 @@ mod consts {
 mod consts {
     use libc::c_int;
```
Comment (Susurrus) [inline src/fcntl.rs]: This needs to `use libc::self` here as well, same as for NetBSD. We just don't test on DragonFly. If you want to install the DragonFly target to test this, that'd be appreciated.

**Comment by Susurrus** [comment]
I'd like to include a change to the `libc_bitflags` to use the same syntax as the new `bitflags` macro. This way we don't need to change `struct` to `flags` and `;` to `,` for every entry. Would you be willing to expand this PR to include a change to that macro here as well? 

Additionally, each of these `const` modules can be combined into one and you can use `#cfg[]` attributes to specify which platforms have which constants. This way we only have each variable declared once instead of multiple times for the various platforms. Should also reduce the LoC a decent amount.

**Comment by equal-l2** [comment]
>I'd like to include a change to the libc_bitflags to use the same syntax as the new bitflags macro. This way we don't need to change struct to flags and ; to , for every entry. Would you be willing to expand this PR to include a change to that macro here as well?

That's too hard for me at this moment, sorry.

File: src/fcntl.rs [hunk 2]
```diff
@@ -255,182 +255,53 @@ mod consts {
...
+            O_NOFOLLOW,
+            O_CREAT,
+            O_TRUNC,
+            O_EXCL,
+            O_NOCTTY,
+            O_DIRECTORY,
+            O_CLOEXEC,
+            O_FSYNC,
+            O_NDELAY,
+            #[cfg(any(target_os = "netbsd", target_os = "openbsd", target_os = "macos", target_os = "ios"))]
```
Comment (Susurrus) [inline src/fcntl.rs]: Please wrap lines so they're under 100 characters.

File: src/fcntl.rs [hunk 3]
```diff
@@ -255,182 +255,53 @@ mod consts {
...
+            O_ALT_IO,
+            #[cfg(target_os = "netbsd")]
+            O_NOSIGPIPE,
+            #[cfg(target_os = "netbsd")]
+            O_SEARCH,
         }
     );
 
-    bitflags!(
+    libc_bitflags!(
```
Comment (Susurrus) [inline src/fcntl.rs]: You already have `FdFlag` declared at the top, so one of these declarations should be removed.

Comment (equal-l2) [inline src/fcntl.rs]: `FdFlag` exists line 242 for linux & android, and line 303 for BSDs.
It needs to merge linux's `consts` with BSDs one to unify `FdFlags`.

Comment (Susurrus) [inline src/fcntl.rs]: Yeah, it'd be nice to merge all of these `consts` modules together into one with `cfg` directives for every constant that needs it. It should be a small amount of work given that you've already merged the BSDs.

Comment (equal-l2) [inline src/fcntl.rs]: Some of linux flags are not in `libc` yet, so I can't use `libc_bitflags!` for them now.
This makes it impossible to merge all `consts` into one.

Comment (Susurrus) [inline src/fcntl.rs]: Yep, that makes sense. We'll defer that until upstream includes those constants.

**Pull request merged**